### ISSUE TITLE
Question about PC softmax implementation

### ISSUE THREAD

**Issue by aiexperience** [hyperconnect/LADE#12]
Hi, I was really impressed with your research. Your code really helped my toy problem experiment and it works well. But I'm a bit confused about matching your paper and code.

 I understood that PC softmax term (LADE-CE loss) is implemented in "loss/PriorCELoss.py". In the paper, equation 4 said that PC Strategy considers source and target probability. ( -log p_s(y) + log p_t(y)  ) But in the "loss/PriorCELoss.py", it only considers prior (source probability) implemented with the function "calculate_prior".

If I misunderstood, it would be of great help to me if you let me know what was wrong.

Thank you!

**Comment by juice500ml** [comment]
Hello, thank you for the interest to our paper!

As you correctly mentioned, LADE-CE loss has to add target probability (log p_t(y)). However, [when we evaluate after training](https://github.com/hyperconnect/LADE/blob/510c03e/run_networks.py#L609-L623), we assume uniform prior, so we don't need to add target probability (i.e. Table 2, 3, 4, 5 in our paper).

But for test time shifted cases, we've experimented with non-uniform prior (Table 6, 9). For that purpose, we [stored logits](https://github.com/hyperconnect/LADE/blob/510c03e/run_networks.py#L557-L560) so that we can load them and apply different target priors just before evaluation. I want to emphasize that this is one of the advantages of using PC softmax, where the model trained with it can adapt to any target prior easily without retraining. We have to simply add log prior before evaluation!

**Comment by aiexperience** [comment]
Thanks for your reply, your answer was enough to understand the question.
